ctapipe/ml/tools/apply.py

```diff
@@ -0,0 +1,255 @@
+
+from ctapipe.core.tool import Tool
+from ctapipe.core.traits import Bool, Path, create_class_enum_trait, flag
+from ctapipe.io import TableLoader, write_table
+from ctapipe.io.tableio import TelListToMaskTransform
+
+from ..apply import EnergyRegressor, ParticleIdClassifier
+from ..sklearn import Classifier, Regressor
+from ..stereo_combination import StereoCombiner
+
+
+class Apply(Tool):
+
+    overwrite = Bool(default_value=False).tag(config=True)
+
+    input_url = Path(
+        default_value=None,
+        allow_none=False,
+        directory_ok=False,
+        exists=True,
+    ).tag(config=True)
+    output_path = Path(
+        default_value=None,
+        allow_none=False,
+        directory_ok=False,
+    ).tag(config=True)
+
+    energy_regressor_path = Path(
+        default_value=None,
+        allow_none=True,
+        exists=True,
+        directory_ok=False,
+    ).tag(config=True)
+    particle_classifier_path = Path(
+        default_value=None,
+        allow_none=True,
+        exists=True,
+        directory_ok=False,
+    ).tag(config=True)
+
+    stereo_combiner_type = create_class_enum_trait(
+        base_class=StereoCombiner,
+        default_value="StereoMeanCombiner",
+    ).tag(config=True)
+
+    aliases = {
+        ("i", "input"): "Apply.input_url",
+        "regressor": "Apply.energy_regressor_path",
+        "classifier": "Apply.particle_classifier_path",
+        ("o", "output"): "Apply.output_path",
+    }
+
+    flags = {
+        **flag(
+            "overwrite",
+            "Apply.overwrite",
+            "Overwrite tables in output file if it exists",
+            "Don't overwrite tables in output file if it exists",
+        ),
+        "f": (
+            {"Apply": {"overwrite": True}},
+            "Overwrite output file if it exists",
+        ),
+    }
+
+    classes = [
+        TableLoader,
+        Regressor,
+        Classifier,
+        StereoCombiner,
+    ]
+
+    def setup(self):
+        """"""
+        self.log.info("Copying to output destination.")
+        shutil.copy(self.input_url, self.output_path)
+
+        self.h5file = tables.open_file(self.output_path, mode="r+")
+        self.loader = TableLoader(
+            parent=self,
+            h5file=self.h5file,
+            load_dl1_images=False,
+            load_dl1_parameters=True,
+            load_dl2=True,
+            load_simulated=True,
+            load_instrument=True,
+        )
+
+        if self.energy_regressor_path is not None:
+            self.regressor = EnergyRegressor.read(
+                self.energy_regressor_path,
+                self.loader.subarray,
+                parent=self,
+            )
+            self.regressor_combine = StereoCombiner.from_name(
+                self.stereo_combiner_type,
+                combine_property="energy",
+                algorithm=self.regressor.model.model_cls,
+                parent=self,
+            )
+            self.apply_regressor = True
+        else:
```

Comment:
Ah bit nice code is to have:

```
self.apply_regressor = False
self.apply_classifier = False
```
at the top and then get rid of the else branch.